components/service/pocket/src/main/java/mozilla/components/service/pocket/NetworkUseCases.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket
+
+import mozilla.components.concept.fetch.Client
+
+/**
+ * Possible actions for usecases which need to access the Internet.
+ */
+internal interface NetworkUseCases {
+    /**
+     * HTTP client used for network requests.
+     * Can be set through [initialize] or unset through [reset].
+     */
+    var fetchClient: Client?
+
+    /**
+     * Convenience method for setting the the HTTP Client which will be used
+     * for all REST communications with the Pocket server.
+     */
+    fun initialize(client: Client)
+
+    /**
+     * Convenience method for cleaning up any resources held for communicating with the Pocket server.
+     */
+    fun reset()
+}
+
+/**
+ * Base class for usecases which need to access the Internet.
+ * Allow to easily set and get a [Client] reference through static methods.
+ */
+@Suppress("UtilityClassWithPublicConstructor")
+internal open class NetworkUseCase {
```

Comment:
This doesn't quite follow A-Cs design in respect to use cases. If you look at our `UseCases` classes, they all contain a list of concrete use case implementation e.g., `DefaultRemoveTabUseCases` in `TabsUseCases`.

I guess the goal here is to share code between the `PocketStoriesUseCase` and the new `SpocsUseCase`. Maybe an abstract class would be better? I got confused about the interface and the companion object implementing it. I think the name should also be `PocketNetworkUseCase`.

Reply:
In the previous implementation, for just recommendations only I went with a companion object and static properties instead of usecases instances to be able to easily use those details inside WorkManager's [Worker which doesn't have a constructor in which to pass the usecase](https://cs.android.com/androidx/platform/frameworks/support/+/79a471d7d4788be5c5e73929daf02a5736f26f43:work/work-runtime-ktx/src/main/java/androidx/work/PeriodicWorkRequest.kt).

Now indeed I used the interface just to sweeten the companion objects and use delegation there to share some functionality between the now two similar usecases.

Seems like WorkManager does allow for a custom factory though, I'll look into that for passing the usecases as arguments and avoiding to rely on static properties.

Reply:
Sounds good, thanks!

Reply:
Looked into that approach.
Code would look nicer - https://gist.github.com/Mugurell/fa1f67d7ee40744cb41312c221b4ca98
but we cannot really use a custom factory as the WorkManager is a singleton instantiated when the app starts and that is the moment where the custom factory to inject custom dependencies would need to be set. - https://developer.android.com/topic/libraries/architecture/workmanager/advanced/custom-configuration#on-demand
For this Pocket stories usecase and generally all our current WorkManager usages in individual AC components I don't see how we could do that.

So my initial approach of having the worker know where from / how to ask for dependencies (reading them from static properties) seems like the most we can do if the worker cannot be constructed / injected with the intended dependencies.
Will do the naming change.

Reply:
OK, looked it again. I think your design is good, and it's really just the naming.